lib/thread.c

```diff
@@ -126,8 +126,6 @@ M0_INTERNAL struct m0_thread *m0_thread_self(void)
 
 M0_INTERNAL int m0_thread_adopt(struct m0_thread *thread, struct m0 *instance)
 {
-	M0_PRE(M0_IS0(thread));
-
 	return m0_thread_arch_adopt(thread, instance, true);
```

Comment:
Because `m0_thread_adopt()` panics on it:

```
Jul 06 00:06:50 centos79 hare-hax[54165]: motr[54165]:  8a60  FATAL  [lib/assert.c:50:m0_panic]  panic: (({ unsigned __nr = (sizeof *(threa
Jul 06 00:06:50 centos79 hare-hax[54165]: Motr panic: (({ unsigned __nr = (sizeof *(thread)); unsigned i; for (i = 0; i < __nr && ({ ((char
Jul 06 00:06:50 centos79 hare-hax[54165]: 2021-07-06 00:06:50,420 [DEBUG] {MainThread} Starting m0_halon_interface
Jul 06 00:06:50 centos79 hare-hax[54165]: /home/ant/cortx-motr/motr/.libs/libmotr.so.2(m0_arch_backtrace+0x20)[0x7f7d9840f8d0]
Jul 06 00:06:50 centos79 hare-hax[54165]: /home/ant/cortx-motr/motr/.libs/libmotr.so.2(m0_arch_panic+0xe6)[0x7f7d9840fa86]
Jul 06 00:06:50 centos79 hare-hax[54165]: /home/ant/cortx-motr/motr/.libs/libmotr.so.2(+0x36bf64)[0x7f7d983fef64]
Jul 06 00:06:50 centos79 hare-hax[54165]: /home/ant/cortx-motr/motr/.libs/libmotr.so.2(m0_thread_adopt+0x2c)[0x7f7d98405c9c]
Jul 06 00:06:50 centos79 hare-hax[54165]: /home/ant/cortx-motr/motr/.libs/libmotr.so.2(m0_halon_interface_thread_adopt+0x33)
```

Reply:
It seems working now. Restored.